File: packages/shared/src/types/product.ts
import { Category } from './category.js';

export interface Product {
    id: string;
    organizationId: string;
    categoryId: string | null;
    name: string;
    description: string | null;
    price: number;
    stock: number;
    isActive: boolean;
    imageUrl: string | null;
    deletedAt: Date | null;
    createdAt: Date;
    updatedAt: Date;
    category?: Category | null;
    _count?: {
        orderItems: number;
    };
}

export interface CreateProductRequest {
    name: string;
    description?: string;
    price: number;
    stock?: number;
    categoryId?: string;
    isActive?: boolean;
    imageUrl?: string;
}

export type UpdateProductRequest = Partial<CreateProductRequest>;

File: apps/ios-pos/pos/pos/Models/Product.swift
import Foundation
import SwiftData

@Model
final class CachedProduct {
    @Attribute(.unique) var id: String
    var organizationId: String
    var categoryId: String
    var name: String
    var productDescription: String? // description 回避
    var price: Int
    var isAvailable: Bool
    var sortOrder: Int
    
    var category: CachedCategory?
    
    init(id: String, organizationId: String, categoryId: String, name: String, productDescription: String? = nil, price: Int, isAvailable: Bool = true, sortOrder: Int = 0) {
        self.id = id
        self.organizationId = organizationId
        self.categoryId = categoryId
        self.name = name
        self.productDescription = productDescription
        self.price = price
        self.isAvailable = isAvailable
        self.sortOrder = sortOrder
    }
}

// MARK: - API DTO

/// バックエンドの GET /organizations/:orgId/products レスポンス用 DTO
struct APIProduct: Decodable {
    let id: String
    let organizationId: String
    let categoryId: String?
    let name: String
    let price: Int
    let stock: Int
    let isActive: Bool
}
